astropy/cosmology/core.py

```diff
@@ -263,6 +263,13 @@ def __init__(self, H0, Om0, Ode0, Tcmb0=2.725, Neff=3.04,
             if self._massivenu:
                 nu_y = self._massivenu_mass / (kB_evK * self._Tnu0)
                 self._nu_y = nu_y.value
+                # This may seem strange, but it turns out to
+                # be much faster to use _nu_y as a python list than
+                # an ndarray for scalar evaluations; see
+                # _massivenu_relative_density_scalar.  So we provide
+                # both -- the ndarray for when z is an ndarray, and
+                # the list version for inside the integrals
+                self._nu_ylist = self._nu_y.tolist()
```

Comment:
In other words (and I tested this) if you had say `Neff=1000` all with different masses, using a Numpy ufunc in `_massivenu_relative_density_scalar` was about 2 orders of magnitude faster.  But when you get down to tiny arrays of 3 or 4 elements the overhead of setting up the ufunc and returning arrays overtakes.